RPIPs/RPIP-47.md

```diff
@@ -1,46 +0,0 @@
----
-rpip: 47
-title: Enable Forced Delegate Upgrades
-description: Allows protocol governance to force upgrade Node Operators after a Rocket Pool protocol upgrade takes place, and a grace period has expired. 
-author: Valdorff (@Valdorff)
-discussions-to: TBD
-status: Draft
-type: Protocol
-category: Core
-created: 2024-03-08
-requires: 43
-tags: tokenomics-2024, tokenomics-content
----
-
-## Abstract
-Currently, Node Operators can choose to upgrade (or not) for their minipool delegate (the contract at the Ethereum withdrawal address that governs how funds are disbursed, among other things). This (a) means the protocol can only effect change in ways that are in the interest of individual NOs, and (b) makes it more challenging to design the protocol as it must remain backwards compatible with every past minipool delegate.
-
-This proposal suggests limiting that upgrade choice in the future. Users will be able to opt in to upgrade or use an "old" delegate for a defined period after a newer version. However, once the defined period has passed, they will no longer use the old delegate. To ensure timely availability of protocol funds, after the defined period, anyone may change the megapool's delegate to a minimal 'recovery delegate' that is only capable of exiting validators and distributing funds. 
-
-## Specification
-### Megapool delegates
-- `Megapool delegate` contracts SHALL have an expiration block (ie, execution layer block)
-- When a new `megapool delegate` is released, the contract upgrade SHALL include:
-  - Setting the new `megapool delegate`'s expiration block to "no expiration"
-  - Setting the previous `megapool delegate`'s expiration block to occur `delegate_upgrade_buffer` after the upgrade
-- A node operator SHALL be able to upgrade their active `megapool delegate` to the latest `megapool delegate` at any time (including after the active `megapool delegate`'s expiration block)
-- After the active `megapool delegate`'s expiration block:
-  - The active `megapool delegate` SHALL NOT be usable by the node operator
```

Comment:
Does not-usable need to be better defined? Does this block vote weight? Which functions? etc.

Reply:
I think it's well defined as is? All functions on the megapool delegate that require the NO will be blocked. Actually... one thing that comes to mind is that we should block withdrawal address interactions too.

Voting doesn't use the megapool delegate, so it would still function. As long as money is properly stuck, I think the incentive is in place to fix it.

Reply:
Should it be very explicit? "Functions in the active megapool delegate SHALL revert when not called by a protocol contract"?

Reply:
I'm worried that might be too restrictive. Will try something like what I wrote above